applets/notification_area/status-notifier/sn-item-v0.c

```diff
@@ -256,7 +256,7 @@ update (SnItemV0 *v0)
       if (!pixbuf){
           /*try to find icons specified by path and filename*/
           pixbuf = gdk_pixbuf_new_from_file(v0->icon_name, NULL);
-          if (pixbuf){
+          if (pixbuf && icon_size > 1) {
```

Comment:
All this code seems weird to me:
- icon_name is supposed to be an icon name. why load it as file?
- what is random `-2`? how that goes together with comment? icon is scaled always...
- looking at code what is effective_icon_size? How can icon_size be <= 0?

@vkareh you say that icon_size default value is 1 !?, but in init it is set to 16... What and why sets icon_size to 1? That warning is because you have 1x1 icon and you try to scale it to -1x-1, because of that random number `2`...

Reply:
Correct, I can see the `icon_size = MAX (1, ...);` above, so that means I could probably just skip this entire chunk of code based on the `icon_size`. I'll look at `effective_icon_size` - it seems to be only set if `icon_size` is `<= 0`, and then it's used to replace `icon_size` itself during size allocation - so this code is not necessarily super clean. I'll work on that.

I don't know about loading `icon_name` as a file, I agree that's strange, but in my tests it seems to be used and working (it actually loads an icon)